homework/index.js

```diff
@@ -29,12 +31,115 @@
+  function addCont(login, img, contributeNumber, parent, link) {
+    const tr = createAndAppend('tr', parent);
+    const imageColumn = createAndAppend('th', tr);
+    createAndAppend('img', imageColumn, {
+      src: img,
+      class: 'contributorImg',
+    });
+    const name = createAndAppend('td', tr);
+    createAndAppend('a', name, {
+      href: link,
+      target: '_blank',
+      text: login,
+    });
+    const contributeNamePlace = createAndAppend('td', tr);
+    createAndAppend('span', contributeNamePlace, {
+      text: contributeNumber,
+      class: 'contributeNumber',
+    });
+  }
+
+  function addRow(property, key, parent) {
+    const tr = createAndAppend('tr', parent);
+    createAndAppend('th', tr, {
+      text: `${property} :`,
+    });
+    createAndAppend('td', tr, { text: key });
+    return tr;
+  }
+
+  let optionNumber = 0;
+  function madeOption(repo, selectButton) {
+    createAndAppend('option', selectButton, {
+      text: repo.name,
+      value: optionNumber,
+    });
+    optionNumber++;
+  }
+
+  function renderContributorDetails(conts, contributorsContainer) {
+    const table = createAndAppend('table', contributorsContainer);
+    conts.forEach(cont => {
+      addCont(
+        cont.login,
+        cont.avatar_url,
+        cont.contributions,
+        table,
+        cont.html_url,
+      );
+    });
+  }
+
+  function renderRepoDetails(repo, div) {
+    const table = createAndAppend('table', div);
+    const firstRow = addRow('Repository', '', table);
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      target: '_blank',
+      text: repo.name,
+    });
+    addRow('Description', repo.description, table);
+    addRow('Forks', repo.forks, table);
+    addRow('Updated', repo.updated_at, table);
+  }
+
+  function fetchContributors(url, contributorsContainer) {
+    const promiseContributor = fetchJSON(url);
+    promiseContributor.then((conts, err) => {
+      if (err) {
+        createAndAppend('div', contributorsContainer, {
+          text: err.message,
+          class: 'alert-error',
+        });
+        return;
+      }
+      renderContributorDetails(conts, contributorsContainer);
+    });
+  }
+
+  function deleteContainers() {
+    document.getElementById('flexContainer').remove();
+  }
+
+  function makeFlexContainer(root, repos, selectButton) {
+    const flexContainer = createAndAppend('section', root, {
+      id: 'flexContainer',
+    });
+    const repoContainer = createAndAppend('section', flexContainer, {
+      class: 'container',
+    });
+    const contributorsContainer = createAndAppend('section', flexContainer, {
+      class: 'container',
+    });
+    createAndAppend('h5', contributorsContainer, { text: 'Contributions' });
+    // eslint-disable-next-line func-names
+    repos
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach(repo => {
+        madeOption(repo, selectButton);
+      });
```

Comment:
You are re-adding `<option>` elements each time a user changes the selected repository. For instance, I changed the selection from the first repo to the second repo, and now the select element contains these repeated option elements:

```html
<select>
  <option value="0">alumni</option>
  <option value="1">angular</option>
  <!-- etc -->
  <option value="45">TicTacToeTDD</option>
  <option value="46">ukulele</option>
  <option value="0">alumni</option>
  <option value="1">angular</option>
  <!-- etc -->
  <option value="45">TicTacToeTDD</option>
  <option value="46">ukulele</option>
</select>
```

Instead of creating the option elements here, you need to do it once only, directly after you have received the repository information from the GitHub API.

Furthermore, when you do that, there is not much to be gained from putting the code to create an `<option>` element in a separate function. In addition, you can use the `index` parameter to set the value of each option element like this:

```js
.forEach((repo, index) => {
  createAndAppend('option', selectButton, {
    text: repo.name,
    value: index,
  });
});
```

Reply:
This problem is still not solved: every time a user changes the selection, all repository names are again added to the select element, which thereby grows each time, as you can see in the figure below. Note that after `46 ukelele` it starts again with `0 alumni`.

![image](https://user-images.githubusercontent.com/2788771/65815129-6afc6200-e1eb-11e9-8536-5017514751ea.png)